### PR TITLE
Fix antlion guard dynamic prop shoving/punting from stock HL2

### PR DESCRIPTION
**Pull request by Blixibon**
In Half-Life 2, antlion guards are supposed to be able to punt props at targets they can't reach. However, this works by making antlion guards "see" certain props when they spawn, and that isn't compatible with the AI sensing code, which requires sight targets to be alive.

This simply makes an exception in the AI sensing code for "objects" (i.e. things that wouldn't be alive) and hooks the dynamic punting behavior to cvars so that the min/max mass can be controlled, or the behavior can be disabled if needed. The upper-limit for mass has also been raised so that antlion guards can punt heavier props (e.g. Arbeit cars).

---

I mainly restored this to try to make antlion guards spawned via Xen grenades more effective (since the environment they were spawned in might not let them reach certain targets), although I had trouble spawning guards in locations where there were props around to punt, mainly since most nearby props were been consumed by the Xen grenades which spawned them.

Maybe I should put this in Mapbase instead 🥴

### CODE REVIEW

File: sp/src/game/server/ai_senses.cpp
Comment (1upD) [inline sp/src/game/server/ai_senses.cpp:148]: How can we be sure this won't have unintended side effects?

Comment (Blixibon) [reply sp/src/game/server/ai_senses.cpp:148]: `FL_OBJECT` is not assigned by and/or used by anything else except for missiles and, in E:Z2, item crates. Neither of them use `IsAlive()` in any functional capacity, except the latter overrides it to allow item crates to be seen by the sensing code. (which could theoretically be removed with this code in place)

Comment (1upD) [reply sp/src/game/server/ai_senses.cpp:148]: I haven't forgotten about this! I want to do a little bit of a deep dive into this tomorrow just for my own understanding. Once I've looked through the code and I feel like I understand it better, I'll approve this PR

Comment (1upD) [reply sp/src/game/server/ai_senses.cpp:148]: I've noticed that the drivable APC also is given the flag ``FL_OBJECT``. So, after this change, projectiles like flares, grenades, etc as well as drivable APCs and remarkables will pass "ShouldSeeEntity" when they would have failed before. Will this have any effect? I think that we should test with the APC after merging this PR. Just to make sure that NPCs don't act differently around projectiles, vehicles, etc. I don't know exactly how this function is used so I'm just trying to be cautious.

**Review by 1upD — APPROVED** [review]
Approving this PR to merge, but after the next build is made we should retest some Venture Ghost maps to make sure there are no side effects of the change to CAI_Senses::ShouldSeeEntity()

**Pull request merged**
